ember-server/shared/src/main/scala/org/http4s/ember/server/internal/WebSocketHelpers.scala

```diff
@@ -164,13 +164,17 @@ object WebSocketHelpers {
       write: Pipe[F, Byte, Unit],
       frameTranscoder: FrameTranscoder,
       closeState: Ref[F, Close],
+      filterPings: Boolean,
   )(implicit F: Concurrent[F]): Pipe[F, WebSocketFrame, WebSocketFrame] = {
     def writeFrame(frame: WebSocketFrame): F[Unit] =
       Stream(frame).through(encodeFrames(frameTranscoder)).through(write).compile.drain
 
     stream =>
       stream.evalMapFilter[F, WebSocketFrame] {
-        case WebSocketFrame.Ping(data) => writeFrame(WebSocketFrame.Pong(data)).as(None)
+        case WebSocketFrame.Ping(data) =>
+          val pong = writeFrame(WebSocketFrame.Pong(data))
+          if (filterPings) pong.as(None)
+          else pong.as(WebSocketFrame.Ping(data).some)
```

Comment:
... but the only changes we need are here, to align with the Blaze implementation.
https://github.com/http4s/http4s/blob/6e13209e8b7ae6d2369ff7c32cc26c1b31d9e35e/blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala#L126-L128

Reply:
Aha, makes sense.  Thanks for the feedback! Made the update 👍🏻

Reply:
Fantastic, thanks!